guava-compatibility-agent/build.gradle

```diff
@@ -3,6 +3,7 @@ apply plugin: 'com.palantir.shadow-jar'
 
 dependencies {
     shadeTransitively 'net.bytebuddy:byte-buddy'
+    compileOnly 'com.google.guava:guava:32.1.2-jre'
```

Comment:
out of curiosity, what's the reason we only needed to add this now? I guess previously this code only accessed guava types via bytebuddy at runtime?

Reply:
This compile-only dependency is used to implement `FallbackAdapterHandler` which delegates to `Futures.catchingAsync` as well as `FallbackAdapter` which implements `AsyncFunction<Throwable, T>`.

That could be done using byte-buddy to define classes, but it's a fair bit trickier.